### PR TITLE
leap: Make exercise schema-compliant

### PR DESCRIPTION
**Pull request by catb0t**
Related to #625.

### CODE REVIEW

File: exercises/leap/canonical-data.json [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
-         "input": 2000,
-         "expected": true
-      }
-   ]
+  "exercise": "leap",
+  "version": "1.0.0",
+  "comments": [
+    "Given a Gregorian calendar year, indicate whether it is a leap year."
+  ],
+
```
Comment (rbasso) [inline exercises/leap/canonical-data.json]: This comment was not in the original test suite and also is not needed for compliance, so maybe it should not be part of this commit/PR.

File: exercises/leap/canonical-data.json [hunk 2]
```diff
@@ -1,24 +1,34 @@
...
+  "cases": [
+    {
+       "description": "year not divisible by 4: common year",
+       "property": "notLeap",
```
Comment (rbasso) [inline exercises/leap/canonical-data.json]: #### About `isLeap` and `notLeap`

All the tests in this test suite use exactly the same test logic, only changing the test data, so they should have the same `property` value. `isLeapYear`and `leapYear` are good candidates

**Comment by catb0t** [comment]
Oh, ok -- hopefully this is more what you wanted.
